src/toxcore/onion/packet/onion_request_2.rs

```diff
@@ -38,15 +38,15 @@ pub struct OnionRequest2 {
 
 impl FromBytes for OnionRequest2 {
     named!(from_bytes<OnionRequest2>, do_parse!(
-        verify!(rest_len, |len| len <= ONION_MAX_PACKET_SIZE) >>
+        verify!(rest_len, |len| *len <= ONION_MAX_PACKET_SIZE) >>
         tag!(&[0x82][..]) >>
         nonce: call!(Nonce::from_bytes) >>
         temporary_pk: call!(PublicKey::from_bytes) >>
         rest_len: rest_len >>
-        payload: cond_reduce!(
+        payload: map_opt!(cond!(
```

Comment:
How about `rest_len: verify!(rest_len, |rest_len| rest_len >= ONION_RETURN_2_SIZE)` instead of `cond`+`map_opt`?

Reply:
done